src/main/java/com/yellow/ordermanageryellow/controller/UserController.java

```diff
@@ -1,24 +1,102 @@
+
 @RestController
 @RequestMapping("/User")
 public class UserController {
 
     private final UsersService usersService;
+    private final MapStructMapper mapStructMapper;
+
     @Autowired
-    public UserController(UsersService usersService){
-        this.usersService=usersService;
+    public UserController(UsersService usersService, MapStructMapper mapStructMapper) {
+        this.usersService = usersService;
+        this.mapStructMapper = mapStructMapper;
     }
+
     @GetMapping("/login")
-    public ResponseEntity<String> login(@RequestParam  String password, @RequestParam String email){
-        return usersService.login(email,password);
+    public ResponseEntity<String> login(@RequestParam String password, @RequestParam String email, @RequestHeader String token) {
+        try {
+            return ResponseEntity.ok().body(usersService.login(email, password));
+        } catch (NotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+    }
+
+    @PostMapping()
+    public ResponseEntity<String> createNewUser(@RequestBody Users newUser, @RequestHeader String token) {
+        try {
+
+            usersService.createNewUser(newUser);
+        } catch (ObjectExistException e) {
+            return ResponseEntity.status(HttpStatus.CREATED).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+        return ResponseEntity.ok().body("success");
+    }
+
+    @DeleteMapping(value = "/{id}")
+    public ResponseEntity<String> deleteUser(@PathVariable String id, @RequestHeader String token) {
+        try {
+            usersService.deleteUser(id);
+        } catch (NotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+        return ResponseEntity.status(HttpStatus.OK).body(id);
     }
 
+    @PutMapping()
+    public ResponseEntity updateCustomer(@RequestBody Users user, @RequestHeader String token) {
+        try {
+            usersService.updateUser(user);
+        } catch (NotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+        return ResponseEntity.status(HttpStatus.OK).body("update success " + user.getFullName());
+
+    }
+
+    @GetMapping("/{pageNumber}/{pageSize}")
+    public ResponseEntity<List<UserDTO>> getAllCustomers(@PathVariable int pageNumber, @PathVariable int pageSize, @RequestHeader String token) {

```

Comment:
rename to getUsers
pageSize should be constant in system

Reply:
this api get only customers (by roleId)
to change  it anyway?

Reply:
For users, there are 2 APIs
one - getUsers for "user management" screen with page number
second - getCustomerNames for new order popup